src/test/regress/expected/subqueries_not_supported.out

```diff
@@ -157,6 +157,17 @@ WHERE
 ;
 ERROR:  cannot push down this subquery
 DETAIL:  Having qual without group by on partition column is currently unsupported when a subquery references a column from another query
+-- We do not support GROUPING SETS in subqueries
+-- This also includes ROLLUP or CUBE clauses
+SELECT * FROM (SELECT user_id, value_1 FROM users_table GROUP BY GROUPING SETS ((user_id), (value_1))) s;
```

Comment:
These tests are good for now. But, I've a note here for when you touch similar parts of the code in the future or if you have seen `random()` calls in the tests.

In general, PostgreSQL has the ability to pull up subqueries. So. say we have the following query:
```SQL
SELECT * FROM (SELECT user_id FROM users_table) as foo;
```

It simply converts (e.g., pulls-up the subquery) the query to the following and plans in that form for various performance reasons:
```SQL
SELECT user_id FROM users_table;
```

For that to happen, the subquery should conform to the rules in [`is_simple_subquery()`](https://doxygen.postgresql.org/prepjointree_8c_source.html#l01442).

As you can see, if `subquery->groupingSets` the function returns false, so we're safe in these tests. 

But, in general, to ensure that Postgresql doesn't do that, we add a `random()` function call since there is the following check in the subquery, and that is less likely to change in the future:

```C
 if (contain_volatile_functions((Node *) subquery->targetList))
         return false;
```

So, if I were to write these queries, I'd ended-up with the following to prevent subquery being pulled:
```SQL

SELECT * FROM (SELECT random() * user_id, value_1 FROM users_table GROUP BY GROUPING SETS ((user_id), (value_1))) s;

```

Reply:
I read the tests in this file to get a general understanding, but I didn't pay attention to random number generation in subqueries.

It is really nice for you to point these things out as it helps me learn how Postgres internals work.
Thanks